File: easy/morse_code_words/morse_code_words.go
package main

import "fmt"

var morseCodes = []string{".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---", "-.-", ".-..", "--", "-.", "---", ".--.", "--.-", ".-.", "...", "-", "..-", "...-", ".--", "-..-", "-.--", "--.."}

func uniqueMorseRepresentations(words []string) int {
	transformations := make(map[string]bool)

	for _, word := range words {
		morseCode := ""
		for _, letter := range word {
			morseCode += morseCodes[letter-'a']
		}
		transformations[morseCode] = true
	}
	return len(transformations)
}

func main() {
	fmt.Println("Solution 1 : ", uniqueMorseRepresentations([]string{"gin", "zen", "gig", "msg"}))
}

File: easy/array/morse_code_words/morse_code_words.py
class Solution(object):
    def uniqueMorseRepresentations(self, words):
        """
        :type words: List[str]
        :rtype: int
        """
        morse_codes = [".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---", "-.-", ".-..",
                       "--", "-.", "---", ".--.", "--.-", ".-.", "...", "-", "..-", "...-", ".--", "-..-", "-.--", "--.."]

        transformations = {}
        for word in words:
            morse_code = ""
            for letter in word:
                morse_code += morse_codes[ord(letter) - ord('a')]
            transformations[morse_code] = True

        return len(transformations)


s = Solution()
print("Solution 1 : ", s.uniqueMorseRepresentations(
    ["gin", "zen", "gig", "msg"]))
